pkg/sql/tablemetadatacache/testdata/update_table_metadata_errors

```diff
@@ -0,0 +1,309 @@
+1 51 system public span_count 2 0 {} 0 0 0 0 0 error1 2021-01-07 06:13:20 +0000 UTC TABLE {"auto_stats_enabled": null, "stats_last_updated": null}
+1 52 system public privileges 5 2 {} 0 0 0 0 0 error1 2021-01-07 06:13:20 +0000 UTC TABLE {"auto_stats_enabled": null, "stats_last_updated": null}
+1 53 system public external_connections 7 0 {} 0 0 0 0 0 error1 2021-01-07 06:13:20 +0000 UTC TABLE {"auto_stats_enabled": null, "stats_last_updated": null}
+1 54 system public job_info 4 0 {} 0 0 0 0 0 error1 2021-01-07 06:13:20 +0000 UTC TABLE {"auto_stats_enabled": null, "stats_last_updated": null}
+1 55 system public span_stats_unique_keys 2 1 {} 0 0 0 0 0 error1 2021-01-07 06:13:20 +0000 UTC TABLE {"auto_stats_enabled": null, "stats_last_updated": null}
+1 56 system public span_stats_buckets 5 1 {} 0 0 0 0 0 error1 2021-01-07 06:13:20 +0000 UTC TABLE {"auto_stats_enabled": null, "stats_last_updated": null}
+1 57 system public span_stats_samples 2 1 {} 0 0 0 0 0 error1 2021-01-07 06:13:20 +0000 UTC TABLE {"auto_stats_enabled": null, "stats_last_updated": null}
+1 58 system public span_stats_tenant_boundaries 2 0 {} 0 0 0 0 0 error1 2021-01-07 06:13:20 +0000 UTC TABLE {"auto_stats_enabled": null, "stats_last_updated": null}
+1 59 system public task_payloads 8 0 {} 0 0 0 0 0 error1 2021-01-07 06:13:20 +0000 UTC TABLE {"auto_stats_enabled": null, "stats_last_updated": null}
+1 60 system public tenant_tasks 7 0 {} 0 0 0 0 0 error1 2021-01-07 06:13:20 +0000 UTC TABLE {"auto_stats_enabled": null, "stats_last_updated": null}
+1 61 system public statement_activity 17 7 {} 0 0 0 0 0 error1 2021-01-07 06:13:20 +0000 UTC TABLE {"auto_stats_enabled": null, "stats_last_updated": null}
+1 62 system public transaction_activity 14 7 {} 0 0 0 0 0 error1 2021-01-07 06:13:20 +0000 UTC TABLE {"auto_stats_enabled": null, "stats_last_updated": null}
+1 63 system public tenant_id_seq 1 0 {} 0 0 0 0 0 error1 2021-01-07 06:13:20 +0000 UTC SEQUENCE {"auto_stats_enabled": null, "stats_last_updated": null}
+1 64 system public mvcc_statistics 6 0 {} 0 0 0 0 0 error1 2021-01-07 06:13:20 +0000 UTC TABLE {"auto_stats_enabled": null, "stats_last_updated": null}
+1 65 system public transaction_execution_insights 23 2 {} 0 0 0 0 0 error1 2021-01-07 06:13:20 +0000 UTC TABLE {"auto_stats_enabled": null, "stats_last_updated": null}
+1 66 system public statement_execution_insights 29 4 {} 0 0 0 0 0 error1 2021-01-07 06:13:20 +0000 UTC TABLE {"auto_stats_enabled": null, "stats_last_updated": null}
+1 67 system public table_metadata 18 10 {} 0 0 0 0 0 error1 2021-01-07 06:13:20 +0000 UTC TABLE {"auto_stats_enabled": null, "stats_last_updated": null}
+
+
+set-time unixSecs=1710000000
+----
+
+update-cache
+----
+updated 57 table(s)
+
+# Now the last_update_error column should be nil.
+query
+SELECT
+  db_id,
+  table_id
+  db_name,
+  table_name,
+  store_ids,
+  total_ranges,
+  last_updated,
+  last_update_error
```

Comment:
nit: should we include all the deterministic columns here, like we do in the non-failure case query to ensure that everything is updated correctly after an error

Reply:
Done.